cannot open source file

After deleting my bin/obj folders, I get this when trying to build:

`c1 : fatal error C1083: Cannot open source file: '<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\BC3.g.c': No such file or directory`

I've seen this with multiple projects (I've created new projects multiple times when encountering this problem).

This is an x86 project and does have

`<RuntimeIdentifier>win-x86</RuntimeIdentifier>`

set per #58 and [this commit](https://github.com/AaronRobinsonMSFT/DNNE/pull/67/commits/61bf700313d91a407ff9bc5e18f2ad19cb2c5e09) .

Yes, I'm using Visual Studio 2022 Community, and I installed the latest x64 and x86 .NET 6 SDKs this morning.

I manually deleted bin/obj to create a clean slate build while debugging other issues.

Here's the build log for a Rebuild All as it hits this project:

`Rebuild started...
2>------ Rebuild All started: Project: BC3, Configuration: Debug Any CPU ------
2>Generator: 'obj\Debug\net6.0-windows\win-x86\BC3.dll':
2>Method 'TestGetStr' has non-exportable type 'String'
2>Building native export: "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\bin\Hostx64\x86\cl.exe" /Od /LDd /TC /MT /GS /Zi /D DNNE_ASSEMBLY_NAME=BC3 /D DNNE_COMPILE_AS_SOURCE /I "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\include" /I "C:\Users\<userid>\.nuget\packages\dnne\1.0.32\tools\platform" /I "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x86\6.0.10\runtimes\win-x86\native" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt"  "<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\BC3.g.c" "C:\Users\<userid>\.nuget\packages\dnne\1.0.32\tools\platform\platform.c" /link /DLL /LTCG /LIBPATH:"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\lib\x86" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.19041.0\um\x86" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.19041.0\ucrt\x86" "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x86\6.0.10\runtimes\win-x86\native\libnethost.lib" Advapi32.lib /IGNORE:4099 /IMPLIB:"<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\bin\BC3NE.lib" /OUT:"<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\bin\BC3NE.dll"
2>Microsoft (R) C/C++ Optimizing Compiler Version 19.31.31104 for x86
2>Copyright (C) Microsoft Corporation.  All rights reserved.
2>
2>BC3.g.c
2>c1 : fatal error C1083: Cannot open source file: '<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\BC3.g.c': No such file or directory
2>platform.c
2>Generating Code...
2>C:\Users\<userid>\.nuget\packages\dnne\1.0.32\build\DNNE.targets(154,5): error MSB3073: The command ""C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\bin\Hostx64\x86\cl.exe" /Od /LDd /TC /MT /GS /Zi /D DNNE_ASSEMBLY_NAME=BC3 /D DNNE_COMPILE_AS_SOURCE /I "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\include" /I "C:\Users\<userid>\.nuget\packages\dnne\1.0.32\tools\platform" /I "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x86\6.0.10\runtimes\win-x86\native" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\shared" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um" /I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\ucrt"  "<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\BC3.g.c" "C:\Users\<userid>\.nuget\packages\dnne\1.0.32\tools\platform\platform.c" /link /DLL /LTCG /LIBPATH:"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.31.31103\lib\x86" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.19041.0\um\x86" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.19041.0\ucrt\x86" "C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x86\6.0.10\runtimes\win-x86\native\libnethost.lib" Advapi32.lib /IGNORE:4099 /IMPLIB:"<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\bin\BC3NE.lib" /OUT:"<path>\BC3\obj\Debug\net6.0-windows\win-x86\dnne\bin\BC3NE.dll" " exited with code 2.
2>Done building project "BC3.csproj" -- FAILED.
`

It may be worth adding that I was just creating a number of test methods to verify basic behavior, and in so doing I noted that at present there doesn't seem to be a way to return a string or directly accept an object passed in as a COM reference. I'm assuming I can work around the reference object issue using something like IntPtr but not sure about the string return.

--- a/sample/Exports.cs
+++ b/sample/Exports.cs
@@ -21,7 +21,7 @@ using System.Runtime.InteropServices;
 
 public class Exports
 {
-    [UnmanagedCallersOnlyAttribute(EntryPoint = "FancyName")]
+    [UnmanagedCallersOnly(EntryPoint = "FancyName")]
     public static int MyExport(int a)
     {
         return a;

--- a/sample/Sample.csproj
+++ b/sample/Sample.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
     <EnableDynamicLoading>true</EnableDynamicLoading>
   </PropertyGroup>
 

--- a/sample/Sample.csproj
+++ b/sample/Sample.csproj
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="DNNE" Version="1.0.32" />
+    <PackageReference Include="DNNE" Version="1.0.33" />
   </ItemGroup>
 
 </Project>

--- a/src/dnne-pkg/dnne-pkg.csproj
+++ b/src/dnne-pkg/dnne-pkg.csproj
@@ -17,7 +17,7 @@
 
   <PropertyGroup>
     <PackageId>DNNE</PackageId>
-    <Version>1.0.32</Version>
+    <Version>1.0.33</Version>
     <Authors>AaronRobinsonMSFT</Authors>
     <Owners>AaronRobinsonMSFT</Owners>
     <Description>Package used to generated native exports for .NET assemblies.</Description>

--- a/src/msbuild/DNNE.BuildTasks/Windows.cs
+++ b/src/msbuild/DNNE.BuildTasks/Windows.cs
@@ -97,7 +97,7 @@ namespace DNNE.BuildTasks
             }
 
             // Set linker flags
-            linkerFlags.Append($"/DLL /LTCG ");
+            linkerFlags.Append($"/DLL ");
 
             if (!string.IsNullOrEmpty(exportsDefFile))
             {

--- a/src/msbuild/DNNE.targets
+++ b/src/msbuild/DNNE.targets
@@ -60,7 +60,7 @@ DNNE.targets
 
   <Target
     Name="DnneGenerateNativeExports"
-    Condition="'$(DnneGenerateExports)' == 'true'"
+    Condition="('$(DesignTimeBuild)' != 'true' OR '$(BuildingProject)' == 'true') AND '$(DnneBuildExports)' == 'true'"
     Inputs="@(IntermediateAssembly)"
     Outputs="@(DnneGeneratedSourceFile)"
     AfterTargets="CoreCompile">

